modules/combined/examples/periodic_strain/global_strain_pfm.i

```diff
@@ -0,0 +1,364 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 50
+  ny = 50
+  xmin = -0.5
+  xmax = 0.5
+  ymin = -0.5
+  ymax = 0.5
+[]
+
+[MeshModifiers]
+  [./cnode]
+    type = AddExtraNodeset
+    coord = '0.0 0.0'
+    new_boundary = 100
+  [../]
+  [./anode]
+    type = AddExtraNodeset
+    coord = '0.0 0.5'
+    new_boundary = 101
+  [../]
+[]
+
+[Variables]
+  [./u_x]
+  [../]
+  [./u_y]
+  [../]
+  [./global_strain]
+    order = THIRD
+    family = SCALAR
+  [../]
+  [./c]
+    [./InitialCondition]
+      type = RandomIC
+      min = 0.49
+      max = 0.51
+    [../]
+  [../]
+  [./w]
+  [../]
+[]
+
+[AuxVariables]
+  [./dispg_x]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./dispg_y]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./local_energy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./disp_x]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./disp_y]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[AuxKernels]
+  [./ug_x]
+    type = GlobalDisplacementAux
+    variable = dispg_x
+  [../]
+  [./ug_y]
+    type = GlobalDisplacementAux
+    variable = dispg_y
+    component = 1
+  [../]
+  [./u_x]
+    type = TotalDisplacementAux
+    variable = disp_x
+    displacement_variables = 'u_x dispg_x'
+  [../]
+  [./u_y]
+    type = TotalDisplacementAux
+    variable = disp_y
+    displacement_variables = 'u_y dispg_y'
+  [../]
+  [./local_free_energy]
+    type = TotalFreeEnergy
+    block = 0
+    execute_on = 'initial LINEAR'
+    variable = local_energy
+    interfacial_vars = 'c'
+    kappa_names = 'kappa_c'
+  [../]
+
+[]
+
+[GlobalParams]
+  derivative_order = 2
+  enable_jit = true
+  displacements = 'u_x u_y'
+  block = 0
+[]
+
+[Kernels]
+  [./TensorMechanics]
+  [../]
+
+  # Cahn-Hilliard kernels
+  [./c_dot]
+    type = CoupledTimeDerivative
+    variable = w
+    v = c
+    block = 0
+  [../]
+  [./c_res]
+    type = SplitCHParsed
+    variable = c
+    f_name = F
+    kappa_name = kappa_c
+    w = w
+    block = 0
+  [../]
+  [./w_res]
+    type = SplitCHWRes
+    variable = w
+    mob_name = M
+    block = 0
+  [../]
+[]
+
+[ScalarKernels]
+  [./global_strain]
+    type = GlobalStrain
+    variable = global_strain
+    global_strain = appl_stress
+  [../]
+[]
+
+[BCs]
+  [./Periodic]
+    [./disp1]
+      auto_direction = 'x'
+      variable = 'u_x'
+    [../]
+    [./disp2]
+      auto_direction = 'y'
+      variable = 'u_y'
+    [../]
+    [./up_down]
+      primary = top
+      secondary = bottom
+      translation = '0 -1 0'
+      variable = 'c w'
+    [../]
+    [./left_right]
+      primary = left
+      secondary = right
+      translation = '1 0 0'
+      variable = 'c w'
+    [../]
+  [../]
```

Comment:
The above periodic BCs can be simplified to

```
  [./Periodic]
    [./all]
      variable = 'u_x u_y c w'
      auto_direction = 'x y'
    [../]
  [../]
```

Reply:
Yes, I am cross-checking the solutions for different problems to make sure the approach is right.